src/lib/Components/Inbox/Conversations/Messages.tsx

```diff
@@ -120,11 +123,21 @@ export class Messages extends React.Component<Props, State> {
 
     return (
       <FlatList
-        inverted
-        data={messages}
+        inverted={!this.state.shouldStickFirstMessageToTop}
+        data={this.state.shouldStickFirstMessageToTop ? messages.reverse() : messages}
         renderItem={this.renderMessage.bind(this)}
         onEndReached={this.loadMore.bind(this)}
         onEndReachedThreshold={0.2}
+        onContentSizeChange={(width, height) => {
+          // If there aren't enough items to scroll through
+          // display messages from the top
+          const windowHeight = Dimensions.get("window").height
```

Comment:
We have had issues in the past with `Dimensions` and iPad rotation; I try to use the [`onLayout` pattern](https://github.com/artsy/emission/blob/master/src/lib/Containers/WorksForYou.tsx#L103) when possible because it's more reliable. Just a thing to look out for once we start making an iPad-ready MVP.

Reply:
Yeah, I was reluctant to use `onLayout` in this case because I would need to wait for both callbacks (`onLayout` and `onContentSizeChange`) to make the computations. Hopefully the latest react native addressed the issues with `Dimensions`

Reply:
Do you mean that there have been changes to the implementation?